### PR TITLE
feat: add set scrolling & get scrollInfo & better freeze API for facade

### PR DESCRIPTION
**Pull request by lumixraku**
close #xxx

Add scrolling API for facade.   And add a more  comprehensible API for freezing command.

packages/sheets-ui/src/facade/f-worksheet.ts
onScroll(callback: (params: Nullable<IViewportScrollState>) => void):
getScrollState(): Nullable<IScrollState>
scrollToCell(row: number, column: number): void

packages/sheets/src/facade/f-worksheet.ts
freezeRow(startRow: number, endRow: number) => void
freezeColumn(startColumn: number, endColumn: number) => void

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/sheets/src/facade/f-worksheet.ts [hunk 1]
```diff
@@ -455,7 +455,7 @@ export class FWorksheet extends FBase {
     }
 
     /**
-     * Hides one or more consecutive rows starting at the given index. Use 0-index for this method.
+     * Hide one or more consecutive rows starting at the given index. Use 0-index for this method.
```
Comment (VicKun4937) [inline packages/sheets/src/facade/f-worksheet.ts:458]: Unnessceary change, its valid using Hides

File: packages/sheets-ui/src/facade/f-worksheet.ts [hunk 2]
```diff
@@ -85,6 +88,93 @@ export class FWorksheetSkeletonMixin extends FWorksheet implements IFWorksheetSk
...
+        const renderManagerService = this._injector.get(IRenderManagerService);
+        const render = renderManagerService.getRenderById(unitId);
+        if (!render) return null;
+        const scm = render.with(SheetScrollManagerService);
+        return scm.getScrollStateByParam({ unitId, sheetId });
+    }
+
+    /**
+     * Invoked when scrolling the sheet.
+     * @param callback
```
Comment (VicKun4937) [inline packages/sheets-ui/src/facade/f-worksheet.ts:160]: please add the param introduction and example in comments

File: packages/engine-render/src/components/sheets/sheet-skeleton.ts [hunk 3]
```diff
@@ -502,10 +527,13 @@ export class SpreadsheetSkeleton extends Skeleton {
...
-    calculate(bounds?: IViewportInfo): Nullable<SpreadsheetSkeleton> {
+    /**
+     * Refresh cache after markDirty by SheetSkeletonManagerService.reCalculate()
+     * @param bounds
+     */
+    calculate(): Nullable<SpreadsheetSkeleton> {
         this._resetCache();
-
-        this.setStylesCache(bounds);
+        this._updateLayout();
```
Comment (VicKun4937) [inline packages/engine-render/src/components/sheets/sheet-skeleton.ts:536]: does this one maybe cause performance problem?  It seems the sekelton dirty is not equal to layout dirty.  Do we have some state flag to mark the layout change?

**Review by VicKun4937 — APPROVED** [review]

File: packages/engine-render/src/components/sheets/sheet-skeleton.ts (hunk 3 above)
Comment (lumixraku) [reply packages/engine-render/src/components/sheets/sheet-skeleton.ts:536]: known performance issue

**Review by wzhudev — APPROVED** [review]

**Review by Dushusir — APPROVED** [review]

File: packages/sheets/src/facade/f-worksheet.ts [hunk 4]
```diff
@@ -1018,7 +1018,10 @@ export class FWorksheet extends FBase {
 
     /**
      * Sets the frozen state of the current sheet.
-     * @param freeze - The freeze object containing the parameters for freezing the sheet.
+     * @param freeze - the scrolling viewport start range and count of freezed rows and columns.
+     * that means if you want to freeze the first 3 rows and 2 columns, you should set freeze as { startRow: 3, startColumn: 2, xSplit: 2, ySplit: 3 }
+     *
+     * @deprecated use `freezeColumn` and `freezeRow` instead.
```
Comment (hexf00) [inline packages/sheets/src/facade/f-worksheet.ts:1024]: Add warning logs to notify the calling user

File: packages/sheets/src/facade/f-worksheet.ts [hunk 5]
```diff
@@ -1044,6 +1047,8 @@ export class FWorksheet extends FBase {
     /**
      * Get the freeze state of the current sheet.
      * @returns The freeze state of the current sheet.
+     *
+     * @deprecated use `getRowFreezeStatus` and `getColumnFreezeStatus` instead.
```
Comment (hexf00) [inline packages/sheets/src/facade/f-worksheet.ts:1051]: Add warning logs to notify the calling user

File: packages/sheets/src/facade/f-worksheet.ts [hunk 6]
```diff
@@ -1103,6 +1108,82 @@ export class FWorksheet extends FBase {
...
+     * ``` ts
+     * const fWorkbook = univerAPI.getActiveWorkbook();
+     * const fWorkSheet = fWorkbook.getActiveSheet();
+     * // freeze the first too columns.
+     * fWorkSheet.freezeColumn(0, 2);
+     * ```
+     * @param startColumn
+     * @param endColumn
+     */
+    freezeColumn(startColumn: number, endColumn: number): void {
```
Comment (hexf00) [inline packages/sheets/src/facade/f-worksheet.ts:1126]: Merge logic using override and setFrozenColumns

File: packages/sheets/src/facade/f-worksheet.ts [hunk 7]
```diff
@@ -1103,6 +1108,82 @@ export class FWorksheet extends FBase {
...
+     *
+     * @example
+     * ``` ts
+     * const fWorkbook = univerAPI.getActiveWorkbook();
+     * const fWorkSheet = fWorkbook.getActiveSheet();
+     * // freeze the first too rows.
+     * fWorkSheet.freezeRow(0, 2);
+     * ```
+     */
+    freezeRow(startRow: number, endRow: number): void {
```
Comment (hexf00) [inline packages/sheets/src/facade/f-worksheet.ts:1153]: Merge logic using override and setFreezeRows

**Review by hexf00 — CHANGES_REQUESTED** [review]

**Review by jikkai — APPROVED** [review]

**Review by hexf00 — APPROVED** [review]

**Review by Jocs — APPROVED** [review]

**Pull request merged**